servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Publisher.java

```diff
@@ -2404,7 +2404,7 @@ public final Publisher<T> retryWhen(boolean terminateOnNextException,
      * @see #repeat(boolean, IntPredicate)
      */
     public final Publisher<T> repeat(IntPredicate shouldRepeat) {
-        return repeat(false, shouldRepeat);
+        return repeat(true, shouldRepeat);
```

Comment:
I've provided a detailed reply on the original PR: https://github.com/apple/servicetalk/pull/2639#discussion_r1253763692

Consider removing `repeat*` overloads that take the boolean and always use `terminateOnNextException == true`.